File: code/test/apitest/py/externalstream.py
#!/usr/bin/env python
#
# Copyright (c) 2005-2009 Jaroslav Gresula
#
# Distributed under the MIT license (See accompanying file
# LICENSE.txt or copy at http://jagpdf.org/LICENSE.txt)
#


import jagpdf
import jag.testlib as testlib
import sys
import os

def test_main(argv=None):
    if None == argv:
        argv = sys.argv
    out_file = os.path.abspath(os.path.join(argv[1], 'basic_extstream.pdf'))
    cfg = testlib.test_config()
    doc = jagpdf.create_stream(testlib.FileStreamOut(out_file), cfg)
    doc.page_start(5.9*72, 3.5*72)
    writer = doc.page().canvas()
    doc.page_end()
    doc.finalize()
    doc = None

if __name__ == "__main__":
    test_main()

File: code/test/apitest/cpp/externalstream.cpp
// Copyright (c) 2005-2009 Jaroslav Gresula
//
// Distributed under the MIT license (See accompanying file
// LICENSE.txt or copy at http://jagpdf.org/LICENSE.txt)
//


#include <stdio.h>
#include <string>
#include "testcommon.h"
using namespace jag;

namespace
{
//
//
//
  void test_main(int, char** argv)
  {
      std::string out_file(argv[1]);
      out_file += "/basic_extstream.pdf";

      pdf::Profile cfg(pdf::create_profile());
      cfg.set("doc.compressed", "0");

      StreamFile stream(out_file.c_str());

      pdf::Document doc(pdf::create_stream(&stream, pdf::Profile()));
      doc.page_start(5.9*72, 3.5*72);
      doc.page_end();
      doc.finalize();
  }
}

int externalstream(int argc, char ** const argv)
{
    return test_runner(test_main, argc, argv);
}


/** EOF @file */
